embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/EmbraceLogService.kt

```diff
@@ -0,0 +1,128 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import io.embrace.android.embracesdk.LogExceptionType
+import io.embrace.android.embracesdk.Severity
+import io.embrace.android.embracesdk.capture.metadata.MetadataService
+import io.embrace.android.embracesdk.internal.utils.Uuid
+import io.embrace.android.embracesdk.session.id.SessionIdTracker
+import io.embrace.android.embracesdk.worker.BackgroundWorker
+import io.opentelemetry.api.common.AttributeKey
+import io.opentelemetry.api.logs.Logger
+import java.util.concurrent.TimeUnit
+
+/**
+ * Creates log records to be sent using the Open Telemetry Logs data model.
+ */
+internal class EmbraceLogService(
+    private val logger: Logger,
+    private val metadataService: MetadataService,
+    private val sessionIdTracker: SessionIdTracker,
+    private val backgroundWorker: BackgroundWorker
+) : LogService {
+
+    override fun log(
+        message: String,
+        severity: Severity,
+        properties: Map<String, Any>?
+    ) {
+        val sessionId = sessionIdTracker.getActiveSessionId()
+        val appState = metadataService.getAppState()
+
+        backgroundWorker.submit {
+            // TBD: Check if config enabled
```

Comment:
It'd be worth looking at how `ConfigGate` works if we want to no-op for the whole service when a config value changes. That avoids having to know about any config value that disables the entire service from everywhere within the service.